Light device trigger test fails intermittently

## The problem
<!-- 
  Describe the issue you are experiencing here to communicate to the
  maintainers. Tell us what you were trying to do and what happened.
-->

This test fails intermittently
```
hass = <homeassistant.core.HomeAssistant object at 0x7f5cf0a79978>, calls = []

    async def test_if_fires_on_state_change_with_for(hass, calls):
        """Test for triggers firing with delay."""
        platform = getattr(hass.components, f"test.{DOMAIN}")
    
        platform.init()
>       assert await async_setup_component(hass, DOMAIN, {DOMAIN: {CONF_PLATFORM: "test"}})
E       assert False

tests/components/light/test_device_trigger.py:183: AssertionError
```
```
INFO     homeassistant.components.light:entity_platform.py:179 Setting up light.test
ERROR    homeassistant.components.light:__init__.py:341 Error parsing light profile from /__w/core/core/tests/testing_config/light_profiles.csv: None
```

FAILED tests/components/light/test_device_trigger.py::test_if_fires_on_state_change_with_for[pyloop]

--- a/tests/components/light/test_device_trigger.py
+++ b/tests/components/light/test_device_trigger.py
@@ -95,6 +95,8 @@ async def test_if_fires_on_state_change(hass, calls):
     platform = getattr(hass.components, f"test.{DOMAIN}")
 
     platform.init()
+    await hass.async_block_till_done()
+
     assert await async_setup_component(hass, DOMAIN, {DOMAIN: {CONF_PLATFORM: "test"}})
     await hass.async_block_till_done()
 
@@ -180,6 +182,8 @@ async def test_if_fires_on_state_change_with_for(hass, calls):
     platform = getattr(hass.components, f"test.{DOMAIN}")
 
     platform.init()
+    await hass.async_block_till_done()
+
     assert await async_setup_component(hass, DOMAIN, {DOMAIN: {CONF_PLATFORM: "test"}})
     await hass.async_block_till_done()
 

